src/Controls/tests/UITests/Tests/Issues/Issue18740.cs

```diff
@@ -12,6 +12,8 @@ public Issue18740(TestDevice device)
 
 		public override string Issue => "Virtual keyboard appears with focus on Entry";
 
+		public override bool ResetMainPage => false; // Requieres a NavigationPage to be able to navigate to every specific case (Entry, Editor, SearchBar).
```

Comment:
Some specific tests requires a NavigationPage as root, so, by default we are reseting the MainPage for every test except in cases where is not possible.

Reply:
If it requires a NavigationPage can the test just use a NavigationPage?

This test is kind of funny because the only reason it's like this is because we didn't have main page resetting. 

Like each of these tests on this one could just be changed to set the main page vs pushing a page. 

```C#
[XamlCompilation(XamlCompilationOptions.Compile)]
[Issue(IssueTracker.Github, 18740, "Virtual keyboard appears with focus on Entry", PlatformAffected.Android)]
public partial class Issue18740NavigationPage : NavigationPage
{
	public Issue18740NavigationPage() : base (new Issue18740())
	{

	}
}

public partial class Issue18740 : ContentPage
{
	public Issue18740()
	{
		InitializeComponent();
	}

	async void OnEntryButtonClicked(object sender, EventArgs e)
	{
		await Navigation.PushAsync(new Issue18740Entry());
	}

	async void OnEditorButtonClicked(object sender, EventArgs e)
	{
		await Navigation.PushAsync(new Issue18740Editor());
	}

	async void OnSearchBarButtonClicked(object sender, EventArgs e)
	{
		await Navigation.PushAsync(new Issue18740SearchBar());
	}
}
```

Reply:
But, under some scenario like porting tests, it is easier to override a property than to modify the example to change the base type of the page, no?

Reply:
hmmmm 
can we cross that bridge once we get this in and then look at some examples with porting tests?